### ISSUE TITLE
Add support for catching async handler errors

### ISSUE THREAD

**Issue by dheeraj-p** [hoangvvo/next-connect#41]
Currently `next-connect` has `onError` option to catch errors thrown by handlers and send relevant response. But it doesn't work with `async` handlers. 
What I would like to have is error thrown from a async handler should get caught in `onError` without have to write `try/catch` blocks and then pass the error to `next(Error)` function.

```
import nextConnect from 'next-connect';
handler = nextConnect({ onError });

handler.use(async (req, res) => {
  throw new Error("Something failed");
});
```

**Comment by hoangvvo** [comment]
@dheeraj-p Thanks for reporting! Fixed in `0.6.2`.

**Comment by dheeraj-p** [comment]
Thank you so much
